docs/general/adapters.md

```diff
@@ -36,6 +36,20 @@ The `Attributes` adapter does not include a root key. It is just the serialized
 
 Use either the `JSON` or `JSON API` adapters if you want the response document to have a root key.
 
+To automatically switch between adapters based on the request, set the adapter to `:mime`
+
+```ruby
+  # In configuration
+  ActiveModelSerializers.config.adapter = :mime
+  
+  # In controller action
+  respond_to do |format|
+    format.json { render json: Model.first }     # Uses the JSON adapter
+    format.json_api { render json: Model.first } # Users the JSON:API adapter
```

Comment:
Really, if you want to render different mime types, each mime type should be registered to its own formatter.  That way, you can do the below, which is the Rails-way IMHO and requires no changes to the codebase. 

```
 # In configuration
ActiveModelSerializers.config.adapter = :json
  
# In controller action
respond_to do |format|
    format.json { render json: model, serializer: ModelSerializer }     # Uses the JSON adapter
    format.jsonapi { render jsonapi: model, serializer: ModelSerializer } # Users the JSON:API adapter
    format.bug { render json: model, serializer: ModelSerializer } # Will raise something about bug not being a known mime type
  end
```

The respond_to block does the content type negotiation for you and usually we use a renderer format with the same name as the negotiated / acceptable mime types.

I think I'm making different assumptions than you, so please continue to discuss.